async-nats/src/options.rs

```diff
@@ -160,6 +167,91 @@ impl ConnectOptions {
         }
     }
 
+    /// Authenticate with a JWT. Requires function to sign the server nonce.
+    /// The signing function is synchronous and should not do any blocking io.
+    ///
+    /// # Example
+    /// ```no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> std::io::Result<()> {
+    /// let seed = "SUANQDPB2RUOE4ETUA26CNX7FUKE5ZZKFCQIIW63OX225F2CO7UEXTM7ZY";
+    /// let kp = nkeys::KeyPair::from_seed(seed).unwrap();
+    /// // load jwt from creds file or other secure source
+    /// async fn load_jwt() -> std::io::Result<String> { todo!(); }
+    /// let jwt = load_jwt().await?;
+    ///
+    /// let nc = async_nats::ConnectOptions::with_jwt(jwt, move |nonce| kp.sign(nonce).unwrap())
+    ///     .connect("localhost").await?;
+    /// # std::io::Result::Ok(())
+    /// # }
+    /// ```
+    pub fn with_jwt<S>(jwt: String, sig_cb: S) -> Self
+    where
+        S: Fn(&[u8]) -> Vec<u8> + Send + Sync + 'static,
+    {
+        ConnectOptions {
+            auth: Authorization::Jwt(
+                jwt,
+                Arc::new(move |nonce| Ok(base64_url::encode(&sig_cb(nonce)))),
+            ),
+            ..Default::default()
+        }
+    }
+
+    /// Authenticate with NATS using a `.creds` file.
+    /// Open the provided file, load its creds,
+    /// and perform the desired authentication
+    ///
+    /// # Example
+    /// ```no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> std::io::Result<()> {
+    /// let nc = async_nats::ConnectOptions::with_credentials("path/to/my.creds").await?
+    ///     .connect("connect.ngs.global").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn with_credentials(path: impl AsRef<Path>) -> io::Result<Self> {
```

Comment:
I've been pretty hesitant to do conversions on the users behalf so far, @Jarema wdyt?

Reply:
Didn't we agree that functions that are irrelevant performance-wise (basically are not called a thousand times, like publish() and subscribe()) can have those, for convenience?

Nonetheless, we're not consistent now.
I would be explicit everywhere and do a follow-up PR soon that touches the whole API in this regard. That would give us a better overview and feel. 

We can swap explicit to non-explicit conversions without breaking anything. We can't the other way around.
I would be explicit and follow up soon.

WDYT @caspervonb ?

Reply:
Has been my thinking, anything that is calling .into, you can drop the into but you can't do the opposite to prefer explicit.

Explicit as the global default, then making conscious decisions for allowing implicit conversions sounds good.

Reply:
fixed